backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButtonColors.kt

```diff
@@ -0,0 +1,156 @@
+package net.skyscanner.backpack.compose.button
+
+import androidx.compose.foundation.interaction.InteractionSource
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.tokens.BpkColors
+import net.skyscanner.backpack.compose.utils.animateAsColor
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+sealed class BpkButtonColors {
+
+  @Composable
+  internal open fun disabledBackgroundColor(): Color =
+    dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint01)
+
+  @Composable
+  internal open fun disabledContentColor(): Color =
+    dynamicColorOf(BpkColor.SkyGrayTint04, BpkColor.BlackTint03)
+
+  @Composable
+  internal open fun rippleColor(): Color =
+    Color.Black.copy(alpha = 0.2f)
+
+  @Composable
+  internal abstract fun backgroundColor(interactionSource: InteractionSource): Color
+
+  @Composable
+  internal abstract fun contentColor(interactionSource: InteractionSource): Color
+
+  object Primary : BpkButtonColors() {
+
+    @Composable
+    override fun backgroundColor(interactionSource: InteractionSource): Color =
+      interactionSource.animateAsColor(
+        default = BpkColor.Monteverde,
+      )
+
+    @Composable
+    override fun contentColor(interactionSource: InteractionSource): Color =
+      dynamicColorOf(BpkColor.White, BpkColor.Black)
+
+  }
+
+  object Secondary : BpkButtonColors() {
+
+    @Composable
+    override fun backgroundColor(interactionSource: InteractionSource): Color =
```

Comment:
I don't like that we have a lot of internal code in public space – it makes reading such code as API reference more difficult. What do you think about moving these to extensions?

Reply:
Yes that's a good point - it'll also expose those colours externally and may be abused. Either extensions or use an enum again - some of the colours are also the same across different styles, so a when statement might be the simpler option for these